### PR TITLE
Fix/installation issue

### PR DESCRIPTION
**Pull request by tomas-stefano**
[Trello Card](https://trello.com/c/D8GANZ5a/688-investigate-and-resolve-editor-installation-problems)

## Context

The problem: The Editor installer freezes when the user doesn’t have Git installed.

So it was considered 4 options which it was invested small timebox to see the advantages and disadvantages.

1. Bundle Git installer within app (the command that we run on the latest user researches together).
       If Git is installed then ok or if not installed .. calls the 3rd party installer.
      The problem lies is to know when the installer has finished.
      We can add a mechanism to every now and then to verify if Git is installed and then proceed.
      I have a worked version for this solution but it feels too much error-prone to me.
      Also it requires installers for each platform that we plan to support.
2.   Bundle binaries. Bundle pretty much anything on the app (including dependencies). Again, it needs binaries for each platform we plan to support. The downside is that the installation script will be complicated because it requires to know the platform and which binaries to use.
Also, it requires further updates to all bundled dependencies for each platform. This is by far the most expensive in terms of maintenance cost for us.
3.  Less comfy but we can check if Git is installed or if not -> add some start-up message or before-download message telling the user that needs software xyz in order to run the application. I also have a worked version for this too. Although if we go to this route can someone pass the message to display to the user. e.g “Git is not installed. Install Git first before you proceed the installation”.
4. Download required stuff on demand. Example - this would mean checking the user’s OS and arch and then just download the required installers or binaries if available. Build the stuff on the user’s machine. This will take a long long time to finish. I didn't strongly consider this option because it is also an error-prone solution.

The chosen options were the 3rd one.

### CODE REVIEW

File: lib/installation.js [hunk 1]
```diff
@@ -65,6 +68,23 @@ async function installDependencies () {
...
+  return new Promise((resolve) => { setTimeout(resolve, ms) })
+}
+
+async function checkGIT () {
+  try {
+    execSync('git --version')
+  } catch ({ message }) {
+    installationLogger.log('Git is not installed.')
+    await app.displayNotification(
+      'Before you can install Form Builder you will first need to install Git.', { phase: 'Install Editor' }
```
Comment (brenetic) [inline lib/installation.js]: do we want to provide a link to instructions somewhere on how to do this? i suppose that opens up difficulty in finding some which covers several platforms . . .

Comment (tomas-stefano) [reply lib/installation.js]: Maybe adding to the runbook guides and link here?

Comment (tomas-stefano) [reply lib/installation.js]: We can have a guide that cover Mac for now.

Comment (brenetic) [reply lib/installation.js]: yeah i think that would be cool 👍

Comment (tomas-stefano) [reply lib/installation.js]: Done! I will create shortly the installation instruction on the guide (the same link that it is committed here).

**Review by brenetic — APPROVED** [review]
🚀

**Pull request merged**